erpnext/stock/doctype/pick_list/pick_list.py

```diff
@@ -37,6 +52,32 @@ def before_submit(self):
 			frappe.throw(_('For item {0} at row {1}, count of serial numbers does not match with the picked quantity')
 				.format(frappe.bold(item.item_code), frappe.bold(item.idx)), title=_("Quantity Mismatch"))
 
+	def before_cancel(self):
+		#update picked_qty in SO Item on cancel of PL
+		for location in self.get('locations'):
+			if location.sales_order_item:
+				self.update_so(location.sales_order_item,0,location.item_code)
+
+	def update_so(self,so_item,picked_qty,item_code):
```

Comment:
missing whitespace? Multiple instances of this. 😅 

for future reference: https://peps.python.org/pep-0008/